src/HTTP/Cookies/CookieWrapper.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+namespace ILIAS\HTTP\Cookies;
+
+use Dflydev\FigCookies\SetCookie;
+
+/**
+ * Class CookieFacade
+ *
+ * Facade class for the FigCookies SetCookie class.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @package ILIAS\HTTP\Cookies
+ * @since   5.2
+ * @version 1.0.0
+ */
+class CookieWrapper implements Cookie {
+
```

Comment:
@shecken: I am not sure about that. Let's look closely at an example:

```
public function withValue($value = null)
{
	return new self($this->cookie->withValue($value));
}
```
$this->cookie is an instance of setCookie and setCookie has the following withValue implementation (see: [SetCookie.php](https://github.com/dflydev/dflydev-fig-cookies/blob/master/src/Dflydev/FigCookies/SetCookie.php)):

```
    public function withValue($value = null)
    {
        $clone = clone($this);
        $clone->value = $value;
        return $clone;
    }
```
so $this->cookie->withValue creates a completely new setCookie and directly passed to the constructor of the new CookieWrapper without changing the source object.

However, I do not see why the constructor is used here. Would it not be more efficient to perform (if I remember correctly, constructors are more expensive than clone):

```
public function withValue($value = null)
{
        $clone = clone($this);
        $clone->cookie = $this->cookie->withValue($value);
        return $clone;
}
```